scripts/code-format.py

```diff
@@ -1,19 +1,67 @@
 #!/usr/bin/env python3
 
 import os
+import sys
+
+# Argument enhancement:  to only run `astyle` on a specified directory, to
+# only include changed source code, these arguments have been added.  If
+# run with no arguments, all the normal directories are examined as before:
+# - /demos/
+# - /examples/
+# - /src/
+# - /tests/
+#
+# Args:
+# If ANY args are specified, ONLY run `astyle` on the specified directories.
+# Any combination can be specified.
+include_demos = True
+include_examples = True
+include_src = True
+include_tests = True
+
+args = sys.argv[1:]
+
+# Have any args been specified?
+if args:
+    include_demos = False
+    include_examples = False
+    include_src = False
+    include_tests = False
+
+    for arg in args:
+        match arg:
+            case "demos":
+                include_demos = True
+            case "examples":
+                include_examples = True
+            case 'src':
+                include_src = True
+            case 'tests':
+                include_tests = False
+            case _:
```

Comment:
I would like to make a request. Would it be possible to avoid using `match ... case` in Python in LVGL scripts for a couple more years? It's a Python 3.10 feature. Users running Ubuntu 20.04 (4 years old) have Python 3.8 by default (i.e. myself) and will not be able to run this script anymore.

Can `if ... elif ... else` be used?

Reply:
Absolutely!  I wasn't aware it would cause problems.  Thank you for enlightening me.

Reply:
Thanks for understanding.

Reply:
> Thanks for understanding.

It's a pleasure and an honor to be working with this team, and part of that is that we are working from different platforms!  I used to be a Unix administrator long ago (1993-95) so am familiar with the caveats of sharing files.  We compromise to make it work for everyone.  :-)